### PR TITLE
support altsCallCredentials in GoogleDefaultChannelCredentials

### PR DESCRIPTION
**Pull request by rockspore**

### CODE REVIEW

**Comment by rockspore** [comment]
No test exists for GoogleDefaultChannelCredentials so I didn't add any but mainly tested the inner logic, which is also very simple.

**Comment by rockspore** [comment]
@ejona86 Aside from this PR, do you mind adding me to grpc-java GitHub team so that I can assign reviewers myself? Thanks.

**Comment by ejona86** [comment]
All you need to do is `@` the people you want to take a look. The more valuable piece is generally to have the CI auto-run, but it did that already. Probably because you have write access to C.

I've given you triage permission, so now you should be able to assign people with issues and PRs and mess with labels.

**Review by matthewstevenson88 — COMMENTED** [review]
LGTM for the ALTS direction. Deferring to @ejona86 for gRPC review.

**Review by ejona86 — APPROVED** [review]
Tell me when you're ready and I'll merge it.

File: alts/src/test/java/io/grpc/alts/DualCallCredentialsTest.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+public class DualCallCredentialsTest {
+
+  @Rule public final MockitoRule mocks = MockitoJUnit.rule();
+
+  @Mock CallCredentials tlsCallCredentials;
+
+  @Mock CallCredentials altsCallCredentials;
+
+  private MethodDescriptor<Void, Void> method =
+      MethodDescriptor.<Void, Void>newBuilder()
```
[the remark this answers is not in the document]
Comment (rockspore) [reply alts/src/test/java/io/grpc/alts/DualCallCredentialsTest.java]: Thanks for the tip. Removed this and using this below.

File: alts/src/main/java/io/grpc/alts/GoogleDefaultChannelCredentials.java [hunk 2]
```diff
@@ -72,16 +73,29 @@ public Builder callCredentials(CallCredentials callCreds) {
...
+      altsCallCredentials = callCreds;
+      return this;
+    }
+
     /** Builds a GoogleDefaultChannelCredentials instance. */
     public ChannelCredentials build() {
       ChannelCredentials nettyCredentials =
           InternalNettyChannelCredentials.create(createClientFactory());
       if (callCredentials != null) {
+        if (altsCallCredentials != null) {
```
[the remark this answers is not in the document]
Comment (rockspore) [reply alts/src/main/java/io/grpc/alts/GoogleDefaultChannelCredentials.java]: Yes your suggestion is much better. I had wanted to make it cleaner but didn't figure out a way as elegant as yours. Thanks!

**Comment by rockspore** [comment]
@ejona86 It should be ready. Thanks a lot for quickly reviewing the PR!

**Review by ejona86 — COMMENTED** [review]
LGTM (approval had been reset)

**Pull request merged**